--- a/meson_options.txt
+++ b/meson_options.txt
@@ -103,3 +103,8 @@
        value : false,
        description : 'Enable GCC static analysis support'
 )
+
+option('tests',
+       type : 'boolean',
+       value : true,
+       description : 'Build test programs')
--- a/meson.build
+++ b/meson.build
@@ -269,12 +269,18 @@
 if get_option('compose')
     subdir('compose/')
 endif
-subdir('tools/')
+if get_option('tools')
+    subdir('tools/')
+endif
 subdir('po/')
-subdir('data/')
+if get_option('tools')
+    subdir('data/')
+endif
 subdir('contrib/')
 subdir('docs/')
-subdir('tests/')
+if get_option('tests')
+    subdir('tests/')
+endif
 if get_option('qt')
     subdir('qt/')
 endif
